specification/0041-hashed-mappings-specification.md

```diff
@@ -0,0 +1,146 @@
+- Hashed base-26 names might be less readable than other name formats.
+  - This is unfortunately hard to quantify. Base-10 hashing would be an alternative, although with longer hashes.
+- A specification of the hashed mappings might make the format inflexible.
+  - Changes in hashed already come with great effects, so any such change should probably go through the RFC process anyway.
+
+## Rationale and Alternatives
+
+- This version of hashed is most likely more stable than Mojmap.
+  - Testing has shown that around 50% of class renames in Mojmap are only repackages,
+    which are avoided thanks to package stripping.
+- Hashed mappings can be fully automated, saving us the work of matching intermediary manually.
+  - We've already evaluated that a fully automated system can update hashed within less than 10 minutes of a new release.
+- When working with layered mappings (e. g. quilt mappings on top of hashed) it is preferable
+  to be able to clearly distinguish between mapped and unmapped code.
+  - Encountering Mojmap names when working with QM could be highly confusing.
+- Unique hashed names provide several benefits:
+  - Easier deobfuscation of stacktraces and other errors (especially when using bots for manual translation).
+  - Allow source remapping from hashed to any other mapping without needing context (no classpath or even full class files needed).
+  - Un-hides fields.
+  - Basically no leaking of mojmap.
+    - Makes it easier to argue about mojmap license.
+    - Would allow using it in "taint free" contexts if it ever comes to it (e.g. yarn)
+- Detection of unobfuscated names is now done by checking whether mappings are identity mappings.
+  - Detection via annotations was attempted, but required too many special cases (constructors, enum values array, etc.)
+  - Single character names in the original mapping have a chance of coinciding with the obfuscated names.
+    - We decided to treat all single character identity mappings as obfuscated
+
+## Prior Art
+
+The first attempt at hashed mappings was outlined in [RFC 19](https://github.com/QuiltMC/rfcs/blob/master/rfc/0019-hashed-mojmap.md).
+This was then implemented in the [mappings-hasher](https://github.com/QuiltMC/mappings-hasher).
+
+Later, an attempt was made to create a [simplified hasher](https://github.com/QuiltMC/mappings-hasher/tree/archive/simplified_hasher),
+by only hashing the mappings without information of the jar.
+However, this idea was scrapped due to the drawbacks of non-unique mapping names.
+
+## Unresolved Questions
+
+- Some of our current tooling might not be fully capable of dealing with these new mappings.
+  However, this is to be fixed in the corresponding tooling, rather than the hashed mappings.
+
+- While this simple format seems to work fine, no extensive testing has been done with the rest of the toolchain.
+  Hashed Mojmap will remain on the snapshot maven until we are confident that the tooling works and this RFC is merged.
```

Comment:
As a I have understood, hashed will be made stable once the RFC is merged, right? If so, the first part of this sentence can be removed

Reply:
I'm leaving this in, merging will be the mark that we are confident.